native/core/src/execution/datafusion/shuffle_writer.rs

```diff
@@ -55,17 +55,25 @@ use datafusion::{
     },
 };
 use datafusion_physical_expr::EquivalenceProperties;
+use futures::executor::block_on;
 use futures::{lock::Mutex, Stream, StreamExt, TryFutureExt, TryStreamExt};
 use itertools::Itertools;
 use simd_adler32::Adler32;
-use tokio::task;
 
 use crate::{
     common::bit::ceil,
     errors::{CometError, CometResult},
 };
 use datafusion_comet_spark_expr::spark_hash::create_murmur3_hashes;
 
+/// The status of appending rows to a partition buffer.
+enum AppendRowStatus {
+    /// The difference in memory usage after appending rows
+    MemDiff(Result<isize>),
```

Comment:
Will this always be an increase in memory? Should this use `usize`?

Reply:
It could be decrease too, if flush happens.